tf2_ros/package.xml

```diff
@@ -30,12 +30,6 @@
   <depend>tf2_msgs</depend>
   <depend>tf2_py</depend>
```

Comment:
GitHub UI won't let me leave comments above this line, but this whole package.xml needs to be cleaned up.

1.  Now that we have no python things in here, the only "buildtool_depend" we need is `ament_cmake`.
1.  There is no direct dependency in this repository on `std_msgs`, so it should be removed.
1.  We may as well make `geometry_msgs` a regular `<depend>`, like the rest of the dependencies.
1.  This package no longer has a dependency on tf2_py.
1.  This package no longer needs pytest as a `test_depend`

Similar cleanups should probably be done in the CMakeLists.txt